server/application/application.go

```diff
@@ -888,7 +882,7 @@ func (s *Server) ListResourceEvents(ctx context.Context, q *application.Applicat
 	if err != nil {
 		return nil, fmt.Errorf("error listing resource events: %w", err)
 	}
-	return list, nil
+	return list.DeepCopy(), nil
```

Comment:
Trying to wrap the `kubernetes.Interface` i think is too large of a task :-p. there are so too many methods.